### ISSUE TITLE
Could not open input file: C:\laragon\etc\apps\LMPA\index.phar

### ISSUE THREAD

**Issue by dionisanchez** [dansleboby/LMPA#2]
Hello, thanks for develop this nice tool.

I'm trying to install it but it throws the error shown on topic when installation starts.

Note that you are ignoring index.phar file in gitignore

Thanks!

**Comment by dansleboby** [comment]
Hello,
Yes because you clone project instand of taking release

Please try with release



https://github.com/dansleboby/LMPA/releases

**Comment by dionisanchez** [comment]
So sorry, my bad, thanks again!

**Comment by dansleboby** [comment]
@dionisanchez if you have some time to help me to fix the tools, I saw errors on bug tracker someone is trying to add php version, I think is coming from you.

two thing I need, the list in: C:/laragon/bin/php/php-7.2.19-Win32-VC15-x64
Because I try to read php.ini but can't find-it mayby it an other name in older version of laragon

I also need the content of "c:/laragon/etc/apache2/mod_php.conf" because I try to find the default version of php in laragon from there

Thanks for help :)

**Comment by dionisanchez** [comment]
Hi,

first of all, I dont know how you know I have those files. They are in a different folder from php, and those are the versions I used to use before trying LMPA. Just surprised and scared.. 😂🤣

I will help you anyway.. Get the files attached to this post.

[php-7.2.19-file-list.csv](https://github.com/dansleboby/LMPA/files/8411732/php-7.2.19-file-list.csv)
[mod_php .txt](https://github.com/dansleboby/LMPA/files/8411741/mod_php.txt)

Here we go..

**Comment by dansleboby** [comment]
The second is generic to laragon

For the first I just get the error from fopen with the path used 😅 it come from bugsnag an error monitoring tool

**Comment by dansleboby** [comment]
"C:\laragon\bin\php_versions_backups\"
Did you move "php-7.2.19-Win32-VC15-x64" inside a backup folder before using LMPA? 

Because the way it working it lookup for the default php version inside mod_php
```
# This file is auto-generated, so please keep it intact.
LoadModule php7_module "C:/laragon/bin/php/php-7.2.19-Win32-VC15-x64/php7apache2_4.dll"
```
Extract the base path of the DLL "C:/laragon/bin/php/php-7.2.19-Win32-VC15-x64" and add /php.ini to get the default value of current php version

So in that case he can't find it because folder no longer exists, I will add a check to seem if path is existing, but this is the issue

Again thanks for help

PS: If you move the folder, you have done this because you are thinking LMPA fully replace php in laragon? I will update the readme to make it more clear

**Comment by dionisanchez** [comment]
> Did you move "php-7.2.19-Win32-VC15-x64" inside a backup folder before using LMPA?

Yes. I had many other PHP versions inside php folder. The installation encountered a php.ini from a 5.30 version despite I picked 7.4 on Laragon and restarted the server and the whole app many times, but LMPA installation kept throwing the same error.

Moving all the php versions but 7.4 out of the folder did the trick.

So, the installation finished seamlessly good. Next step was to install PHP version from the LMPA cli. No issues on that.
Last step I did, converting an app project to LMPA to configure php version.

Now for some reason If I try to start Apache with _aliases selected it throws this error:

![image](https://user-images.githubusercontent.com/84447296/161619351-3ecb69d4-bfa1-428b-9a8d-54240b911e5f.png)

**Comment by dansleboby** [comment]
I should move the _aliases folder, this folder is only used for cli, you add the full path to your system and you got access to php80 php81 php74 and composer80 composer81 composer74 reprensting all version you have add from LMPA

You shloud leave Laragon with a php version like php-7.2.19-Win32-VC15-x64 or php-7.4.19-Win32-vc15-x64 I only test LMPA with 7.4

You can't change the version to other version from LMPA, changing the verison in LMAP will have de default behaviours and cheange it for all app, also using a php downloaded from LMPA is not recommended because it NTS version and it only recommended with FCGI module.


You leave LMPA with 7.4 and in LMPA you choose option 0) Manage PHP version and from there you will be able to change php version per app (vhosts)

**Do you think I should move all php download/aliases outsite the default laragon folder to prevent confusion?**

Thanks for time and help

**Comment by dionisanchez** [comment]
So, if I've understood correctly:

1- Should I leave all the PHP versions that I will need inside the PHP folder?
2- I should have Laragon with version php-7.4.19-Win32-vc15-x64?

If I do this and start LMPA it throws this error:

![image](https://user-images.githubusercontent.com/84447296/161701838-7f113e30-69c6-4125-8553-c8fd073a49e5.png)

Is this correct?

![image](https://user-images.githubusercontent.com/84447296/161704092-80971223-7ca6-4e13-8dbd-5b60a3313f29.png)

I think there is a lack of documentation, if you need help writing it I can try to help you. More extensive documentation with a step-by-step installation method.

**Comment by dansleboby** [comment]
Your right it lak of doc, help will be nice, and I saw more cases and get more details to build a good one, I will try to make one base on your feedback today and I will be more than happy to add more details if you have more.

I saw the issue your are facing I was not thinking of users with multiple php version prior LMPA, I need to update the powershell scripts to find 7.4 in the folder or more, I will try LMPA on php 80 and 81 and mayby look backword to get the newest php. Idealy I would need to pass each php.exe inside each subfolder and check ther version because the folder name does not contain version number.

And Yes your folder structure is right 

I will also try todo an update about this today
